### ISSUE TITLE
Feature Request: Block PRs When Exceeding Size Thresholds

### ISSUE THREAD

**Issue by vmerino** [preactjs/compressed-size-action#107]
Hello, 

Are there any plans to enable blocking the merge of a pull request if it exceeds a specified size threshold? Thanks for your time!

**Comment by rschristian** [comment]
Wouldn't this get quite tedious to constantly bump?

My gut says this is better left as information for people to check while reviewing than a pass/fail action.

**Comment by vmerino** [comment]
Hmm... maybe? Wouldn't it be pass by default, and fail only if threshold is exceeded? If the threshold is set too low, then yes it would be tedious. It seems like some could benefit from having a mechanism for egregious size violations though. For instance, we recently had a case where tree-shaking wasn't occurring as expected, and our bundle size ballooned by 10 MB. In extreme cases like that, I would prefer that the PR be automatically blocked.

**Comment by rschristian** [comment]
> If the threshold is set too low, then yes it would be tedious.

Yeah, this is what I was referring to; I think it'd be quite difficult to have the bar be low enough that's it's often useful while not being a tedious pain.

For example, for a web app, you'd probably set that for 50kb-ish? But then if you need to add some graph lib, or some lib inlines SVGs/CSS/whatever, you'll be back in the action bumping up the size and every time you're adding a sizable feature. To stop it from being a pain, you bump it up to say 1mb, but then that's not realistically useful, as it's only going to catch the most egregious bugs, so you might miss some regressions you wish you hadn't.

There's a lot of nuance to size increases and if they're justified or not which is why I'd say this is better left as a metric for a person to look over, personally.
